zeeguu/core/word_scheduling/basicSR/basicSR.py

```diff
@@ -116,9 +143,63 @@ def find_or_create(cls, db_session, bookmark):
         db_session.commit()
         return b
 
+    @classmethod
+    def priority_bookmarks_to_study(cls, user, required_count):
+        """
+        Prioritizes the bookmarks to study. To randomize the
+        exercise order utilize the Frontend assignBookmarksToExercises.js
+
+        The original logic is kept in bookmarks_to_study as it is called to
+        get similar_words to function as distractors in the exercises.
+
+        Currently, we prioritize bookmarks in the following way:
+        1. Words that are closest to being learned (indicated by `cooling_interval`, the highest the closest it is)
+        2. Words that are most common in the language (utilizing the word rank in the db)
+        """
+
+        def sorting_properties(bookmark):
+            cooling_interval = (
+                cls.query.filter_by(bookmark=bookmark).one().cooling_interval
+            )
+            user_word = UserWord.query.filter_by(id=bookmark.origin_id).one()
+            word_rank = user_word.rank
+            if word_rank is None:
+                word_rank = 10000
```

Comment:
We already have a constant called `UserWord.IMPOSSIBLE_RANK` - should we not use that? 
We might want to rename that, and think about it's value. 
- rename to: `MAX_RANK`? `UNKNOWN_RANK`? keep `IMPOSSIBLE_RANK`? `NO_RANK`? 
- value: 
    1. start using 1,000,000 everywhere? that is clearly future proof, if at some point we want to add more ranks to the DB than 10,000. 
    2. set that constant to 10,000 - we don't have to update the DB to make all the 10,000 into 1MIL... that's not an argument, it's a simple update.